docs/developer-docs/smart-contracts/advanced-features/management-canister.mdx

```diff
@@ -84,14 +84,40 @@ async fn install_code(canister_id: CanisterId, wasm_module: Vec<u8>) {
 <AdornedTab value={"typescript"} label="TypeScript" endAdornment={<BetaChip />}>
 
 ```typescript
-@update([], CreateCanisterResult)
-async executeCreateCanister(): Promise<CreateCanisterResult> {
-    const createCanisterResult = await createCanister();
+import { call, IDL, Principal, query, update } from 'azle';
+import {
+    CanisterInfoArgs,
+    CreateCanisterArgs,
+    CreateCanisterResult,
+} from 'azle/canisters/management';
+
+type State = {
```

Comment:
I would remove all of this `State` stuff, it was for our internal testing and isn't really meant to teach the developer anything.

Reply:
Removed state code